src/GeniusYield/Types/Script.hs

```diff
@@ -830,3 +855,26 @@ instance Eq GYAnyScript where
   GYSimpleScript s1 == GYSimpleScript s2 = s1 == s2
   GYPlutusScript s1 == GYPlutusScript s2 = defaultEq s1 s2
   _ == _                                 = False
+
+instance Ord GYAnyScript where
+  compare (GYSimpleScript s1) (GYSimpleScript s2) = compare s1 s2
+  compare (GYSimpleScript _) (GYPlutusScript _)   = LT
+  compare (GYPlutusScript s1) (GYPlutusScript s2) = defaultCompare s1 s2
+  compare (GYPlutusScript _) (GYSimpleScript _)   = GT
```

Comment:
Is the `Ord` instance really meaningful/necessary?

Reply:
At some places `GYAnyScript` is used as a key in a Map (for instance in `addRefScript` function), etc. It is ofc. not meaningful, but require.

Reply:
Ah, that's quite unfortunate. It's worth revisiting `addRefScript` later to perhaps take a map of pointers to vector approach later on. e.g `indexMap :: Map ScriptHash Int` and `scriptVector :: Vector GYAnyScript`.